### PR TITLE
WIP: add repl command to gatsby-cli

### LINKED ISSUE #6952: Add REPL to Gatsby develop
We've discussed/tried ways to look at gatsby data e.g. redux devtools. But the simplest would be to just allow people to flip to a node repl that has access to all the Gatsby data. This would be fun for checking out data plus running code over data for analysis.

Inspired by https://twitter.com/patrickc/status/1024318691863744512?s=19

### PR DESCRIPTION
**Pull request by kkemple**
I got the itch to experiment with this idea to gauge LOE. I was able to create a basic repl with lots of Gatsby info in global context, however it's really just JSON data and would be hard to work with as is.

I would like to add some globals and commands to make life easier, maybe things like:

- [x] babelrc
- [x] components
- [x] jsonDataPaths
- [x] getNode
- [x] getNodes
- [x] hadNodeChanged
- [X] loadNodeContent
- [x] nodes
- [x] pages
- [x] runQuery
- [x] schema
- [x] (site) config
- [x] static queries
- [ ] create docs

Profit 🎉

This will also need an accompanying docs guide I would imagine. :smile:

Closes #6952 

screen shot 2018-08-12 at 11 20 09 am

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview for *using-contentful* failed.

Built with commit d912e7c0335904aa06b4dadacdfcfdb4d5b639be

https://app.netlify.com/sites/using-contentful/deploys/5b7050483813f024d958f610

File: packages/gatsby/src/commands/repl.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  _.context.babelrc = babelrc
+  _.context.components = components
+  _.context.dataPaths = jsonDataPaths
+  _.context.getNode = getNode
+  _.context.getNodes = getNodes
+  _.context.hasNodeChanged = hasNodeChanged
+  _.context.loadNodeContent = loadNodeContent
+  _.context.nodes = [...nodes.entries()]
+  _.context.pages = [...pages.entries()]
+  _.context.runQuery = query
```
Comment (KyleAMathews) [inline packages/gatsby/src/commands/repl.js]: Could we just name this "graphql" to match the function we pass around internally?

File: packages/gatsby/src/commands/repl.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    prompt: `gatsby > `,
+  })
+
+  // set some globals to make life easier
+  _.context.babelrc = babelrc
+  _.context.components = components
+  _.context.dataPaths = jsonDataPaths
+  _.context.getNode = getNode
+  _.context.getNodes = getNodes
+  _.context.hasNodeChanged = hasNodeChanged
```
Comment (KyleAMathews) [inline packages/gatsby/src/commands/repl.js]: This wouldn't be useful in a repl context.

**Comment by calcsam** [comment]
Made changes @KyleAMathews requests and am going to merge this in so it's not hanging. @kkemple feel free to include docs in a separate PR of course :)

**Pull request merged**